src/pipecat/services/speechmatics/stt.py

```diff
@@ -517,34 +808,56 @@ async def _send_frames(self, finalized: bool = False) -> None:
         if not speech_frames:
             return
 
-        # If final, then re=parse into TranscriptionFrame
+        # Check at least one frame is active
+        if not any(frame.is_active for frame in speech_frames):
+            return
+
+        # Frames to send
+        upstream_frames: list[Frame] = []
+        downstream_frames: list[Frame] = []
+
+        # If VAD is enabled, then send a speaking frame
+        if self._enable_vad and not self._is_speaking:
+            self._is_speaking = True
+            upstream_frames += [BotInterruptionFrame()]
+            downstream_frames += [UserStartedSpeakingFrame()]
```

Comment:
It would be nice to add a debug log here, as we have in the BaseInputTransport:
`logger.debug("User started speaking")`